### PR TITLE
dbeaver/pro#2145 disable encrypt if ssl is not active

### PR DESCRIPTION
**Pull request by Destrolaric**
Closes dbeaver/pro#2145
Test on SQLServer with/without SSL
Test in Cloudbeaver

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/SQLServerDataSource.java [hunk 1]
```diff
@@ -196,7 +197,7 @@ public ServerLoginCache getServerLoginCache() {
     @Override
     protected Properties getAllConnectionProperties(@NotNull DBRProgressMonitor monitor, JDBCExecutionContext context, String purpose, DBPConnectionConfiguration connectionInfo) throws DBCException {
         Properties properties = super.getAllConnectionProperties(monitor, context, purpose, connectionInfo);
-
+        DBWHandlerConfiguration sslHandler = connectionInfo.getHandler(SQLServerConstants.HANDLER_SSL);
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/SQLServerDataSource.java]: I don't see where this variable is used

**Review by LonwoLonwo — APPROVED** [review]

**Review by E1izabeth — APPROVED** [review]

**Comment by MashaKorax** [comment]
verified

File: plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/SQLServerDataSource.java [hunk 2]
```diff
@@ -221,6 +220,8 @@ protected Properties getAllConnectionProperties(@NotNull DBRProgressMonitor moni
         final DBWHandlerConfiguration sslConfig = getContainer().getActualConnectionConfiguration().getHandler(SQLServerConstants.HANDLER_SSL);
         if (sslConfig != null && sslConfig.isEnabled()) {
             initSSL(monitor, properties, sslConfig);
+        } else {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/SQLServerDataSource.java]: User may set this property manually. Test it.

**Review by serge-rider — CHANGES_REQUESTED** [review]

**Review by E1izabeth — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/SQLServerDataSource.java [hunk 3]
```diff
@@ -221,6 +220,8 @@ protected Properties getAllConnectionProperties(@NotNull DBRProgressMonitor moni
         final DBWHandlerConfiguration sslConfig = getContainer().getActualConnectionConfiguration().getHandler(SQLServerConstants.HANDLER_SSL);
         if (sslConfig != null && sslConfig.isEnabled()) {
             initSSL(monitor, properties, sslConfig);
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/SQLServerDataSource.java:222]: Umm??
CommonUtils.isEmpty(PROP_ENCRYPT_SSL is always false.
Please fix and test it

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/SQLServerDataSource.java (hunk 3 above)
Comment (Destrolaric) [reply plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/SQLServerDataSource.java:222]: Fixed

**Comment by MashaKorax** [comment]
verified

**Review by serge-rider — APPROVED** [review]

**Pull request merged**